graph-builder/src/plugins/openshift_secondary_metadata_parser/plugin.rs

```diff
@@ -0,0 +1,513 @@
+                                metadata
+                                    .entry((*key).to_string())
+                                    .and_modify(|previous_add| {
+                                        *previous_add += &format!(",{}", &value)
+                                    })
+                                    .or_insert_with(|| (*value).to_string())
+                            });
+                            true
+                        }
+                        _ => false,
+                    }
+                });
+            })
+        });
+
+        Ok(())
+    }
+
+    async fn process_blocked_edges(&self, io: &mut InternalIO) -> Fallible<()> {
+        let blocked_edges_dir = self.settings.data_directory.join("blocked-edges");
+        let blocked_edges: Vec<graph_data_model::BlockedEdge> =
+            deserialize_directory_files(&blocked_edges_dir, regex::Regex::new("ya+ml")?)
+                .await
+                .context(format!(
+                    "Reading blocked edges from {:?}",
+                    blocked_edges_dir
+                ))?;
+        debug!(
+            "Found {} valid blocked edges declarations.",
+            blocked_edges.len()
+        );
+
+        blocked_edges
+            .into_iter()
+            .try_for_each(|blocked_edge| -> Fallible<()> {
+                let mut to = blocked_edge.to;
+
+                // add build information to match architecture
+                if to.build.is_empty() {
+                    let arch =
+                        // Special case for a few cases where the "s390x" arch was encoded with '-' instead of '+'
+                        if to.pre == vec![semver::Identifier::AlphaNumeric("s390x".to_string())] {
+                            "s390x"
+                        } else {
+                            &self.settings.default_arch
+                        };
+
+                    warn!("Adding architecture {} to {:?}", &arch, &to);
+                    to.build
+                        .push(semver::Identifier::AlphaNumeric(arch.to_string()));
+                };
+
+                // find version in the graph
+                let release_id = match io.graph.find_by_version(&to.to_string()) {
+                    Some(release_id) => release_id,
+                    None => {
+                        warn!("Release with version {} not found in graph", to);
+                        return Ok(());
+                    }
+                };
+
+                // add metadata to block edge using the `previous.remove_regex` metadata
+                io.graph
+                    .get_metadata_as_ref_mut(&release_id)
+                    .context(format!(
+                        "[blocked_edges] Getting mutable metadata for {}",
+                        &to.to_string()
+                    ))?
+                    .insert(
+                        format!("{}.{}", self.settings.key_prefix, "previous.remove_regex"),
```

Comment:
I'm not entirely sure this would be a regex - is it an exact version?

Reply:
If by *this* you mean `blocked_edge.from`, it is defined to be here: https://github.com/openshift/cincinnati/blob/89cd657bd552efdd1f8bd2c3c57a038e6cfecf9d/graph-builder/src/plugins/openshift_secondary_metadata_parser/plugin.rs#L20-L22

Reply:
Oh, correct, thanks!